pkg/registry/cache.go

```diff
@@ -0,0 +1,29 @@
+package registry
+
+import "fmt"
+
+// Cache storage client for an OCI registry.
+type Cache struct {
+	registries map[string]StorageClient
+}
+
+// NewCache creates an OCI registry client.
+func NewCache() *Cache {
```

Comment:
How useful is this cache, given that this is used via the CLI and won't be long-lived?

I imagine we don't care, but this too isn't thread-safe... But unless we decide to parallelize the for loop that's reading from this cache, I suppose it's not important.

Reply:
There is no intention of parallelizing this for now, this is for convenience so we we can mindless loop over the images to copy and reuse a client. The normal case, there are two sources and one destination.

Reply:
oras paralielizes blob (aka image layer) copy under the covers